arch/risc-v/src/common/riscv_internal.h

```diff
@@ -294,6 +294,18 @@ int riscv_check_pmp_access(uintptr_t attr, uintptr_t base, uintptr_t size);
 int riscv_configured_pmp_regions(void);
 int riscv_next_free_pmp_region(void);
 
+/* RISC-V Memorymap Config **************************************************/
+
+static inline void riscv_set_basestack(uintptr_t base, uintptr_t size)
+{
+  unsigned int i;
+
+  for (i = 0; i < CONFIG_SMP_NCPUS; i++)
+    {
+      g_cpux_idlestack[i] = (const uint8_t *)(base + size * i);
```

Comment:
why not initialize the array at the definition place?

Reply:
Initializing this array at the definition sounds like good, but it needs lots of ugly macro test like  `CONFIG_SMP_NCPUS < x`. I don't think it is good.

Reply:
but it still better to runtime initialize it in all chip specific file.

Reply:
I agree, initializing this array statically does have benefits. The only problem is how to achieve it.The only way I find at now is just generating `g_cpux_idlestack` based on the `CONFIG_SMP_NCPUS` . Could you have some better idea?

Reply:
I don't find a better method than this too.:(